src/app/TargetSelect/TargetSelect.tsx

```diff
@@ -85,9 +81,15 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props)
 
   const onSelect = (evt, selection, isPlaceholder) => {
     if (isPlaceholder) {
-      context.target.setTarget('');
+      context.target.setTarget(NO_TARGET);
     } else {
-      if (selection.connectUrl != selected) context.target.setTarget(selection.connectUrl);  
+      if (selection != selected) {
+        try {
+          context.target.setTarget(selection);  
+        } catch (error) {
+          context.target.setTarget(NO_TARGET);
```

Comment:
I guess displaying a notification with the error message might also make sense, so that the user can tell what went wrong? Otherwise, if somehow this case is ever tripped, the user is going to have no idea what went wrong and won't be able to file helpful bug reports for us.

Reply:
Yeah, I was thinking on the one hand the user probably cannot do anything about it, but it makes sense to indicate what went wrong.